### PR TITLE
Ode 1236

### PR DESCRIPTION
**Pull request by mestifanos**
missing.py

This a script to check missing element or a duplicate element in a list.
There is a main function at the bottom for testing.

You can look at both missing_numbers and missing_elements
The script is using missing_numbers

The sample call to the library looks like this 
    def validateSerialNo(self, serialNolist): 
        if serialNolist is not None
            print(list(self.missing_numbers(serialNolist)))
            self.hasduplicate(serialNolist)
        else:
            print("Serial number list is empty.")

### CODE REVIEW

File: validator.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import configparser
```
Comment (mvs5465) [inline validator.py]: This is the library file from the test-harness and is already in this repo: `__init__.py`

Please remove this file.

File: missing.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    print(dup_list)
+
+# num_list is a list of elements such as serial numbers
+def missing_numbers(num_list):
+      original_list = [x for x in range(num_list[0], num_list[-1] + 1)]
+      num_list = set(num_list)
+      return (list(num_list ^ set(original_list)))
+
+# test the methods to check if they are working
+def main():
```
Comment (mvs5465) [inline missing.py]: Is your plan for the validator library to extract a list of recordIds and serialNumbers then pass them to this method? The messages are a list of json strings and this takes a list of integers.

Comment (mestifanos) [reply missing.py]: The plan is for the validator to extract the serial numbers (integers) sent them to the library.

Comment (hmusavi) [reply missing.py]: That's what I was referring to in the meeting. This code relies on sending a list of numbers. So yes, the higher level code, will need to build a list of number and pass it to this method. We need to have that higher level method in the library as well so maintain the same interface.

File: validator.py (hunk 1 above)
Comment (hmusavi) [reply validator.py]: That's because the PR is against master. If we change it to be based on ode-1227, it will be only one file changed.

**Comment by hmusavi** [comment]
These changes were not usable at this time. Functionality was added in a separate branch as part of ODE-1227.

**Pull request closed without merging**